Entities, repos with functions, exception,  FoodController + FoodService

b-melani opened the pull request.

b-melani committed:
Entities, repos with functions, exception,  FoodController + FoodService-t for Post

b-melani committed:
Entities, repos with functions, exception,  FoodController + FoodService-t for Post

borosr commented on src/main/java/com/flow/practice2/entity/User.java:
```diff
@@ -0,0 +1,43 @@
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
Stringen ez alapesetben nem fog menni. Szerintem töröld egyelőre és generálj neki id-t a service-ben.

b-melani replied:
Ez így valid lehet? Mind3 entity ID-jára kell rakni, igaz? 
@Id
@GeneratedValue(generator = "uuid")
@GenericGenerator(name = "uuid", strategy = "uuid2")
private final String id;

borosr commented on src/main/java/com/flow/practice2/exception/ValidationException.java:
```diff
@@ -0,0 +1,11 @@
+import org.apache.catalina.connector.Response;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+public class ValidationException extends Exception{
```
Öröklődjön inkább a RuntimeExceptionből vagy a ResponseStatusException-ből

borosr commented on src/main/java/com/flow/practice2/repository/FoodRepository.java:
```diff
@@ -0,0 +1,21 @@
+import java.util.Optional;
+
+@Repository
+public interface FoodRepository extends PagingAndSortingRepository<Food, String> {
```
Használj inkább JpaRepository-t vagy CrudRepository-t, akkor nem kell override-ollni a metódusokat

borosr commented on src/main/java/com/flow/practice2/repository/OrderRepository.java:
```diff
@@ -0,0 +1,21 @@
+//import java.util.Optional;
+//
+//@Repository
+//public interface OrderRepository extends PagingAndSortingRepository<Order, String> {
```
Ugyanaz, mint a foodnál

borosr commented on src/main/java/com/flow/practice2/service/FoodService.java:
```diff
@@ -0,0 +1,32 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Rakj ki egy @Transactional-t is

borosr commented on src/main/java/com/flow/practice2/service/FoodService.java:
```diff
@@ -0,0 +1,32 @@
+@Service
+public class FoodService {
+
+    @Autowired
```
Nem fontos az @Autowired, ha van konstruktor

borosr commented on target/classes/application.yaml:
```diff
@@ -0,0 +1,17 @@
+server:
```
A target directory-t rakd gitignore-ba

b-melani committed:
Merge remote-tracking branch 'origin/develop' into develop

borosr committed:
add gitignore

b-melani committed:
Merge pull request #2 from borosr/patch-1

Add gitignore

b-melani committed:
added get end post method to FoodController and FoodService

b-melani committed:
Delete target directory

b-melani committed:
port updated to 8090

b-melani committed:
Merge remote-tracking branch 'origin/develop' into develop

b-melani committed:
delete db folder

borosr commented on src/main/java/com/flow/practice2/repository/FoodRepository.java:
```diff
@@ -0,0 +1,20 @@
+@Repository
+public interface FoodRepository extends JpaRepository<Food, String> {
+
+    <S extends Food> S save(S s);
```
Ezek a függvények nem kellenek, alapból benne vannak a JpaRepository-ban

b-melani committed:
add log.debug to FoodController getFood's method

b-melani committed:
added getFoodById

Created method: getFoodById in FoodController
Created method: giveTheFood in FoodService

b-melani committed:
Created class OrderController

has @RestController and @RequestMapping

nomansnick committed:
controller_very_basic_no_happiness

b-melani committed:
.

b-melani committed:
OrderService and OrderController -POST finished

b-melani committed:
Merge pull request #3 from b-melani/Patrik

 123
